script/testScripts/gov/GenerateProposal/GenerateModifyParametersProposal.s.sol

```diff
@@ -0,0 +1,108 @@
+// SPDX-License-Identifier: GPL-3.0
+pragma solidity 0.8.19;
+
+import {JSONScript} from '@script/testScripts/gov/JSONScript.s.sol';
+import {Strings} from '@openzeppelin/utils/Strings.sol';
+import {ODGovernor} from '@contracts/gov/ODGovernor.sol';
+import {Modifiable} from '@contracts/utils/Modifiable.sol';
+import {IModifiable} from '@interfaces/utils/IModifiable.sol';
+import {GenerateProposal} from '@script/testScripts/gov/GenerateProposal.s.sol';
+import 'forge-std/StdJson.sol';
+import 'forge-std/console2.sol';
+
+contract GenerateModifyParametersProposal is GenerateProposal, JSONScript {
+  using stdJson for string;
+
+  error UnrecognizedDataType();
+
+  string public objectKey = 'MODIFY_PARAMS_OBJECT_KEY';
+
+  address[] internal _targets;
+  string[] internal _datas;
+  string[] internal _params;
+  string[] internal _dataTypes;
+
+  address internal _governanceAddress;
+  string internal _description;
+
+  function _loadBaseData(string memory json) internal override {
+    _description = json.readString(string(abi.encodePacked('.description')));
+    _governanceAddress = json.readAddress(string(abi.encodePacked('.odGovernor')));
+    uint256 len = json.readUint(string(abi.encodePacked('.numberOfModifications')));
```

Comment:
since it all parses into one long string, I'm not sure how to get the length.  I tried different ways and settled on the numberOfModifications after a few failed attempts.  but I can try some more and see if there's a better way.

Reply:
added a check to the js script that automatically sets the number of modifications.